crates/rpc/rpc-builder/tests/it/http.rs

```diff
@@ -97,8 +97,8 @@ where
     EthApiClient::send_transaction(client, transaction_request).await.unwrap_err();
     EthApiClient::hashrate(client).await.unwrap();
     EthApiClient::submit_hashrate(client, U256::default(), H256::default()).await.unwrap();
-    EthApiClient::gas_price(client).await.unwrap();
-    EthApiClient::max_priority_fee_per_gas(client).await.unwrap();
+    EthApiClient::gas_price(client).await.unwrap_err();
+    EthApiClient::max_priority_fee_per_gas(client).await.unwrap_err();
```

Comment:
what happened here?

Reply:
Since we wanted to bound the number of requests to `get_block_values` by `min(blocknum, configured_max)`, this commit https://github.com/paradigmxyz/reth/pull/2600/commits/bcdc4048131b2043a461176803e637a2db57177e changed the call for the latest header from this:
```rust
        let header_hash = self
            .client
            .block_hash_for_id(BlockId::Number(BlockNumberOrTag::Latest))?
```
to this:
```rust
        let block = self
            .client
            .block_by_id(BlockId::Number(BlockNumberOrTag::Latest))?
```
When `Latest` is passed to `block_hash_for_id`, it uses the `BlockNumProvider::chain_info()` method:
https://github.com/paradigmxyz/reth/blob/6e15d72e7db698ed8ac8eb6182bd013219012715/crates/storage/provider/src/traits/block_id.rs#L70-L77

The `NoopProvider` implementation uses a default value for this, which always exists, so the call never fails:
https://github.com/paradigmxyz/reth/blob/6e15d72e7db698ed8ac8eb6182bd013219012715/crates/storage/provider/src/test_utils/noop.rs#L31-L36

The default value for the `ChainInfo` struct is the same as the default hash for `GasPriceOracleResult`, so we used to return early and did not error:
https://github.com/paradigmxyz/reth/blob/cc601dbe9f1451b39caa96f6cd35ffe0699dae35/crates/rpc/rpc/src/eth/gas_oracle.rs#L106-L109

The new code uses `block_by_id`, which returns `Ok(None)`, and ends up in an error `UnknownBlockNumber`:
https://github.com/paradigmxyz/reth/blob/6e15d72e7db698ed8ac8eb6182bd013219012715/crates/storage/provider/src/test_utils/noop.rs#L65-L67